ledger/participant-integration-api/src/test/suite/scala/platform/store/backend/DbDtoToStringsForInterningSpec.scala

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.backend
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class DbDtoToStringsForInterningSpec extends AnyFlatSpec with Matchers {
+
+  it should "select all relevant strings for interning" in {
+    val iterators = DbDtoToStringsForInterning(fixture)
+    iterators.templateIds.toList.sorted shouldBe List(
+      "14",
```

Comment:
Should the behavior for duplicated keys be tested? W

Reply:
Only saw your comment after hitting the `like` button. You are absolutely right there. Concatenation of iterators over sets is not the same as iterator of concatenation of sets. Te answer depends whether the client code can live with duplicates.

Reply:
duplicates very much expected here, this logic targets the enumeration of all.
RawStringInterning.newEntries has the .distinct which you are looking for.
sorted list are compared here because iterator of set-s are spawn